test/Concurrency/Runtime/actor_assert_precondition_executor_checkIsolated.swift

```diff
@@ -0,0 +1,92 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-build-swift -Xfrontend -disable-availability-checking -parse-as-library %s -o %t/a.out
+// RUN: %target-codesign %t/a.out
+// RUN:  %target-run %t/a.out
+
+// REQUIRES: executable_test
+// REQUIRES: concurrency
+// REQUIRES: concurrency_runtime
+
+// UNSUPPORTED: back_deployment_runtime
+// UNSUPPORTED: back_deploy_concurrency
+// UNSUPPORTED: use_os_stdlib
+// UNSUPPORTED: freestanding
+
+// rdar://119743909 fails in optimize tests.
+// UNSUPPORTED: swift_test_mode_optimize
+// UNSUPPORTED: swift_test_mode_optimize_size
+
+//import StdlibUnittest
+import Dispatch
+
+// TODO: Dispatch should provide such implementation
+extension DispatchSerialQueue {
+  public func checkIsolated() {
+    dispatchPrecondition(condition: .onQueue(self))
+  }
+}
+
+/// We only do the executor dance because missing 'asUnownedSerialExecutor'
+/// on DispatchSerialQueue on some platforms.
+final class NaiveQueueExecutor: SerialExecutor {
+  let queue: DispatchSerialQueue
+
+  init(queue: DispatchSerialQueue) {
+    self.queue = queue
+  }
+
+  public func enqueue(_ unowned: UnownedJob) {
+    queue.sync {
+      unowned.runSynchronously(on: self.asUnownedSerialExecutor())
+    }
+  }
+
+  public func asUnownedSerialExecutor() -> UnownedSerialExecutor {
+    UnownedSerialExecutor(ordinary: self)
+  }
+
+  func checkIsolated() {
+    self.queue.checkIsolated()
+  }
+}
+
+
+actor FamousActor {
+  let queue: DispatchSerialQueue
+  let executor: NaiveQueueExecutor
+
+  init() {
+    self.queue = DispatchSerialQueue(label: "MyQueue")
```

Comment:
Can we add a similar test for DispatchWorkloop being used as serial executor?

Reply:
Sure, I can give that a go. 

It's a bit unfortunate that I have to keep writing "wrappers" since the conformances don't ship in old enough OS or on Linux, so such "wrapper executor around X" is the best I can do, rather than directly using the queue or workloop...

but I'll give that a go.

Reply:
TODO: I've not done this bit yet but will follow up on it